lib/driver.js

```diff
@@ -233,6 +237,34 @@ class EspressoDriver extends BaseDriver {
     }
   }
 
+  /**
+   * Unzip the given app path if needed.
+   *
+   * @param {string} appPath The path to app file.
+   * @returns {?string} Retuns the path to an unzipped app file path
+   *                    if the given path was unzipped, otherwise null.
+   */
+  async unzipAppIfNeeded (appPath) {
+    if (path.extname(_.toLower(appPath)) !== ZIP_EXT) {
+      return null;
+    }
+
+    const useSystemUnzipEnv = process.env.APPIUM_PREFER_SYSTEM_UNZIP;
+    const useSystemUnzip = _.isEmpty(useSystemUnzipEnv)
+      || !['0', 'false'].includes(_.toLower(useSystemUnzipEnv));
+    const tmpRoot = await tempDir.openDir();
+    await zip.extractAllTo(appPath, tmpRoot, {useSystemUnzip});
+
+    const globPattern = `**/*.+(${SUPPORTED_EXTENSIONS.map((ext) => ext.replace(/^\./, '')).join('|')})`;
+    const sortedBundleItems = (await fs.glob(globPattern, {
+      cwd: tmpRoot,
+      strict: false,
+    })).sort((a, b) => a.split(path.sep).length - b.split(path.sep).length);
+    const unzippedAppPath = path.resolve(tmpRoot, path.basename(_.first(sortedBundleItems)));
```

Comment:
I'm not quite sure `path.basename` should be used here as the destination file might also be found in a subfolder. We should first verify if any item has been matched across `sortedBundleItems` and then join a relative path to the first one with tmpRoot as a result or throw an exception